### DISCUSSION TITLE
Rokus With AirPlay Are Not Discovered in AirPlay Player Providers

### DISCUSSION THREAD

**Discussion by taydy00** [music-assistant/support#2243] [Feature requests and ideas]
### What version of Music Assistant has the issue?

2.0.0b145 (but has always had this issue)

### What version of the Home Assistant Integration have you got installed?

2024.5.1

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Hey all, I have multiple Roku 4K devices (with AirPlay and HomeKit built in). They are both part of my Apple Home (that I no longer use, but regardless they are still part of it). They show up as AirPlay devices for pretty much any Apple device on my (flat) network, and Home Assistant itself has them as media players. However, Music Assistant doesn't see them as AirPlay devices (but does successfully see my HomePod Mini, so AirPlay devices are working correctly). I read about some devices needing to have encryption enabled, however the Rokus don't even show in Music Assistant for me to enable/disable encryption. Is it possible to add player providers manually via IP?

I did also attempt to add them via the Home Assistant Media Player integration, however Roku disabled third-party "Play on Roku" support so that workaround won't work either. You will probably see this in the logs as well.

### How to reproduce

Try to add AirPlay Player Providers, notice that Rokus are not auto-discovered.

### Music Providers

All - this is unrelated to the issue (I believe). However, I have:

Spotify
Jellyfin (currently down, please ignore in logs)
Local Media

### Player Providers

AirPlay (though again, my HomePod Mini is auto-discovered and works as it should)
Sonos
Chromecast
DLNA

### Full log output

[log.txt](https://github.com/music-assistant/hass-music-assistant/files/15277083/log.txt)

### Additional information

My network is a flat network, and there is only one interface for Home Assistant. The only notable thing about the setup is that I use Tailscale (which runs on Home Assistant). However, all of my Linux boxes (including Home Assistant) are able to see all other devices on the LAN, so I don't believe that to be an issue.

### What version of Home Assistant Core are your running

2024.5.1

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by OzGav** [comment]
Are these Airplay 1 or Airplay 2 devices? Airplay 2  only devices are not supported

**Comment by taydy00** [comment]
Ah, you would be correct, they are AirPlay 2. I misunderstood when it said that "AirPlay 2 devices should receive the AirPlay 1 stream without issues". So this means that we can't send an AirPlay 1 stream via MASS to an AirPlay 2 device?

**Comment by OzGav** [comment]
MA will do that but if the device won't/can't understand it then problem...

**Comment by OzGav** [comment]
If you look forther down on that page you will see Samsung does something weird and their devices don't work and not just a problem for MA

**Comment by taydy00** [comment]
I understand - so there's no way to get MASS to see the AirPlay 2 device and try to send a stream to it? Is there anyway I can manually create a playback provider/device just to see what happens?

**Comment by taydy00** [comment]
Right, for sure, but at least the Samsung devices show up within Music Assistant. I can't even see the Roku devices within Music Assistant, hence why I'm wondering if there's something else at play here?

**Comment by OzGav** [comment]
You can check the device settings. If your mini is on the same subnet then there should be no reason why it doesn't work. Have you tried encryption on/off and compression on/off?

Edit: Apple TVs are a problem as well. They don't work.

**Comment by taydy00** [comment]
Sorry if I wasn't clear - The HomePod Mini works just fine and is discovered by Music Assistant. However, my Rokus (with AirPlay 2) are not discovered by Music Assistant. I can't change the encryption/compression settings for the Rokus until Music Assistant can see them, but it does not discover the Rokus. Ultimately, I'm wondering if I can manually add the Roku devices to Music Assistant

Edit: Everything is on the same subnet as well. It just appears as though Music Assistant isn't seeing the Roku's AirPlay broadcast

**Comment by taydy00** [comment]
> You can check the device settings. If your mini is on the same subnet then there should be no reason why it doesn't work. Have you tried encryption on/off and compression on/off?
> 
> Edit: Apple TVs are a problem as well. They don't work.

Definitely, is it possible the Rokus are implementing AirPlay in a similar way to the Apple TV? I would be a little surprised, but I do suppose it's possible...

**Comment by OzGav** [comment]
Oh of course. I wasn't thinking. No you can't manually add devices. I'll move this to a feature request for now and speak to Marcel in due course. He has mentioned trying to get Airplay 2/ATV devices working. Priority unknown!

**Reply by OzGav** [reply]
@taydy00 Just making sure you have a link to this now

**Reply by taydy00** [reply]
I do, thank you! And thank you for your quick replies!

**Reply by taydy00** [reply]
I should add, my iPad running iOS 12 (very very old, only supports AirPlay 1) is able to see the Rokus as well, so it doesn't seem to be related to AirPlay 2

**Reply by fenster24** [reply]
I am experiencing the same issue. I have a few different Roku devices, most of them the latest generation, including a Roku Streambar Pro. I'm more than willing to share logs and perform any testing that might be needed when a developer has time to look into this.

**Reply by Davis8483** [reply]
Same issue as well, in home assistant my Roku shows up as a homekit device with almost no controls but in music assistant, it won't show up at all.

**Comment by marcelveldt** [comment]
Can somebody of you explore the mdns entries on the network ?
On MacOS you have an app called Discovery which lets you browse all mdns announcements in a tree view.
The one we listen for is the _raop service which is airplay 1 (raop, audio only).

Please check if the Roku is even broadcasting this and if so, paste the output here so I can have a look.

<img width="413" alt="Scherm­afbeelding 2024-09-01 om 21 06 20" src="https://github.com/user-attachments/assets/79f12a9b-2f2d-4359-8549-de4750b1bca4">



As for the Apple TV, we filter that one out on purpose as we cant support its auth yet. Other than that we shouldnt be filtering out any devices so I suspect that the Roku is just omitting some info we expect or something like that

**Reply by Davis8483** [reply]
![image](https://github.com/user-attachments/assets/7b5023e6-a909-49d6-8492-d168a37fb999)
It didn't show up for me. I additionally opened the Homekit menu in the Roku settings.

**Reply by Davis8483** [reply]
![image](https://github.com/user-attachments/assets/dc107d00-6466-4e5f-a880-06f65dac8099)
It does seem to show up when I search for the _airplay service instead of _raop

**Reply by fenster24** [reply]
I get the exact same results. Note that the _raop entries for me all come from AirCast Home Assistant Add-on (they're natively Chromecast Audio devices/groups).

![image](https://github.com/user-attachments/assets/d19e9f60-9250-44b2-80e5-cf57eea7e500)

**Reply by marcelveldt** [reply]
OK, so Roku devices are not publishing that they support RAOP, making this a Roku issue and not MA. We only support RAOP/Airplay 1.

**Reply by Davis8483** [reply]
Are there any plans to support Airplay 2 in the future?

**Reply by OzGav** [reply]
Not in the foreseeable future but never say never
